src/main/java/baritone/pathing/movement/movements/MovementDiagonal.java

```diff
@@ -125,17 +126,22 @@ public static void cost(CalculationContext context, int x, int y, int z, int des
             destWalkOn = destInto;
         } else {
             destWalkOn = context.get(destX, y - 1, destZ);
-            if (!MovementHelper.canWalkOn(context, destX, y - 1, destZ, destWalkOn)) {
+            boolean standingOnABlock = MovementHelper.mustBeSolidToWalkOn(context, x, y-1, z, context.get(x, y-1, z));
+            frostWalker = standingOnABlock && MovementHelper.canUseFrostWalker(context, destWalkOn);
+            if (!MovementHelper.canWalkOn(context, destX, y - 1, destZ, destWalkOn) && !frostWalker) {
```

Comment:
move the `!frostWalker` to first so that it short circuits on the boolean

(this is not a big deal because we already have the IBlockState so this is just going to PrecomputedData which is no biggie but might as well)